### ISSUE TITLE
Coolwatcher crashes due to undefined reference to _ZN6QMutex12lockInternalEv

### ISSUE THREAD

**Issue by XaviDCR92** [Ai-Thinker-Open/GPRS_C_SDK#442]
## 1. SDK version（SDK 版本）

{
csdtk42-linux, obtained from https://github.com/pulkin/csdtk42-linux
}

---

## 2. In what kind of operation problems appear, and how to reproduce the problem ?（什么样的操作步骤问题会出现，是否是稳定复现，如何复现问题？）

{
When coolwatcher is executed, it prints `libcoolhost_library.so: undefined reference to _ZN6QMutex12lockInternalEv` over stderr and crashes.
}

---
**Root cause of the problem: libcoolhost_library.so requires Qt4.8.7 but 4.7.4 is distributed with csdtk42-linux.**

`c++filt _ZN6QMutex12lockInternalEv` returns `QMutex::lockInternal()`, where `QMutex` is defined by `QtCore` (as stated [on the documentation](https://doc.qt.io/archives/qt-4.8/qmutex.html)). `libQtCore.so.4.7.4` is distributed along with `csdtk42-linux`, so I looked up for that symbol on the shared object with no success. [According to the documentation](https://github.com/pulkin/csdtk42-linux/blob/master/cooltools/DOC/Coolwatcher_User_Guide_N.docx), `libqt4-qt3support` is required. Unfortunately, Ubuntu 20.04 dropped support for Qt4 on their repositories, so the following third-party PPA must be added:
```
sudo add-apt-repository ppa:rock-core/qt4
sudo apt install libqt4-qt3support
```
Once the package and its dependencies are installed, the symbolic links used by coolwatcher must now point to the 4.8.7 shared objects. I wrote up the following quick dirt bash script to get the job done in `csdtk42-linux/lib`:
```{sh}
#!/bin/bash
for file in `ls libQt*.so.4`; do
    mv $file _$file
    target=/usr/lib/x86_64-linux-gnu/$file.8.7
    echo "TARGET="$target
    ln -s $target $file
done
```
In other words, `sdtk42-linux` should be distributed along with Qt4.8.7 binaries. Since this issue is loosely related to `GPRS_C_SDK`, I'll refer to this post on https://github.com/pulkin/csdtk42-linux too in case anyone is interested.

**Comment by Makimars** [comment]
If anyone comes here now, ermito's pull request should work or I also created an appimage that I shared on my fork of the original repo.

https://github.com/Makimars/GPRS_C_SDK_Linux_coolwatcher_appimage/releases/tag/appimage
